slatedb/src/error.rs

```diff
@@ -146,11 +197,256 @@ impl From<object_store::Error> for SlateDBError {
 /// This enum encapsulates various error conditions that may arise
 /// when parsing or processing database configuration options.
 #[non_exhaustive]
-#[derive(Error, Debug)]
-pub enum SettingsError {
-    #[error("Unknown configuration file format: {0}")]
+#[derive(ThisError, Debug)]
+pub(crate) enum SettingsError {
```

Comment:
Do we still want this as a separate type vs. rolling it into SlateDbError? (could be a follow-on)

Reply:
It might make sense to merge, I'll do it.